pkg/utils/common.go

```
@@ -259,21 +261,27 @@ func LoadEnvFile(fs v1.FS, file string) (map[string]string, error) {
 	return envMap, err
 }
 
-// GetUpgradeTempDir returns the dir for storing upgrade related temporal files
+// GetTempDir returns the dir for storing related temporal files
 // It will respect TMPDIR and use that if exists, fallback to try the persistent partition if its mounted
 // and finally the default /tmp/ dir
-func GetUpgradeTempDir(config *v1.RunConfig) string {
+// suffix is what is appended to the dir name elemental-suffix. If empty it will randomly generate a number
+func GetTempDir(config *v1.RunConfig, suffix string) string {
 	// if we got a TMPDIR var, respect and use that
+	if suffix == "" {
```

I just wonder, why not defaulting to random? Anyway, looks good to me

---

well, for using it on tests its necessary to prefill some stuff in order to trigger some tests.

Plus I think its nice in case we want to reuse it down the line, you can get a fixed tempdir but still get the proper place if needed, in case you want to reuse it somehow?